modules/hostap/Kconfig

```diff
@@ -0,0 +1,64 @@
+# WPA Supplicant configuration options
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config WIFI_NM_WPA_SUPPLICANT
+	bool "WPA Suplicant from hostap project [EXPERIMENTAL]"
+	select POSIX_CLOCK
+	select NET_SOCKETS
+	select NET_SOCKETS_PACKET
+	select NET_SOCKETPAIR
+	select NET_L2_WIFI_MGMT
+	select WIFI_NM
+	select EXPERIMENTAL
+	help
+	  WPA supplicant as a network management backend for WIFI_NM.
+
+config WIFI_NM_WPA_SUPPLICANT_THREAD_STACK_SIZE
+	int "Stack size for wpa_supplicant thread"
+	default 8192
+
+config NET_SOCKETPAIR_BUFFER_SIZE
```

Comment:
Since we're working directly in the Zephyr tree now - should all those defaults for Kconfigs defined in Zephyr be actually set where the individual entries are defined (default 4096 if WIFI_NM_WPA_SUPPLICANT for example)? This default redefinition works in NCS context, as NCS Kconfig files are processed before Zephyr (AFAIK Kconfig will apply the first default it finds), but is it guaranteed to work if we provide custom defaults in Zephyr itself? @tejlmand What's your say on this?

Reply:
I temporarily moved various defaults to their respective Kconfig file. I think we need to revisit those values as I suspect we can tweak them more.
Also I am hoping to get rid of socketpair use between hostap and its handler thread. As they running in the same "process", we can use more lightweight IPC like k_fifo. This would mean that we could most probably use the default values already supplied in zephyr Kconfig files (guessing here) without any need to tweak those. Anyway, that is in future PRs to change.